src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -39,17 +40,26 @@
 /* controls use of locks to enforce safe commutativity */
 bool AllModificationsCommutative = false;
 
+static uint64 shardForTransaction = INVALID_SHARD_ID;
+static bool callbackInstalled = false;
```

Comment:
should probably reference 'xact' or something in th evarname, there could be many differen types of callbacks.